python/ray/tests/test_task_metrics.py

```diff
@@ -0,0 +1,166 @@
+from collections import defaultdict
+import os
+import time
+
+import pytest
+
+import ray
+from ray._private.test_utils import (
+    fetch_prometheus_metrics,
+    wait_for_condition,
+)
+
+
+METRIC_CONFIG = {
+    "_system_config": {
+        "metrics_report_interval_ms": 100,
+    }
+}
+
+
+def tasks_by_state(info) -> dict:
+    metrics_page = "localhost:{}".format(info["metrics_export_port"])
+    print("Fetch metrics from", metrics_page)
+    res = fetch_prometheus_metrics([metrics_page])
+    if "ray_tasks" in res:
+        states = defaultdict(int)
+        for sample in res["ray_tasks"]:
+            states[sample.labels["State"]] += sample.value
+        print("Tasks by state: {}".format(states))
+        return states
+    else:
+        return {}
+
+
+def test_task_basic(shutdown_only):
+    info = ray.init(num_cpus=2, **METRIC_CONFIG)
+
+    @ray.remote
+    def f():
+        time.sleep(999)
+
+    [f.remote() for _ in range(10)]
+
+    expected = {
+        "RUNNING": 2.0,
+        "WAITING_FOR_EXECUTION": 0.0,
+        "SCHEDULED": 8.0,
+        "WAITING_FOR_DEPENDENCIES": 0.0,
+    }
+    # TODO(ekl) optimize the reporting interval to be faster for testing
+    wait_for_condition(
+        lambda: tasks_by_state(info) == expected, timeout=20, retry_interval_ms=2000
```

Comment:
We also don't need retry interval anymore I guess?

Reply:
Good point, let me lower.